app/controllers/api/v2/vcs_clone_controller.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module Api
+  module V2
+    class VcsCloneController < ::Api::V2::BaseController
+      include ::ForemanAnsible::ProxyAPI
+
+      rescue_from ActionController::ParameterMissing do |e|
+        render json: { 'error' => e.message }, status: :bad_request
+      end
+
+      rescue_from Git::GitExecuteError do |_e|
+        head :internal_server_error
+      end
+
+      api :GET, '/vcs_clone/get_repo_info', N_('Returns information about the repo')
+      param :vcs_url, String, N_('Url of the repo'), :required => true
+      error 400, :desc => N_('Parameter unfulfilled')
+      error 500, :desc => N_('Git error')
+      def repo_information
+        vcs_url = params.require(:vcs_url)
+        remote = Git.ls_remote(vcs_url).slice('head', 'branches', 'tags')
+        remote['vcs_url'] = vcs_url
+        render json: remote
+      end
+
+      api :GET, '/vcs_clone/get_installed_roles', N_('Returns an array of roles installed on the provided proxy')
+      formats ['json']
+      param :smart_proxy, Array, N_('Name of the SmartProxy'), :required => true
+      error 400, :desc => N_('Parameter unfulfilled')
+      error 500, :desc => N_('Internal server error')
+      def installed_roles
+        smart_proxy = params.require(:smart_proxy)
+        ansible_proxy = SmartProxy.find_by(name: smart_proxy)
```

Comment:
You should at least require it to have the Ansible feature:
```suggestion
        ansible_proxy = SmartProxy.with_features('Ansible').find_by(name: smart_proxy)
```

But you can also verify a capability:
```suggestion
        ansible_proxy = SmartProxy.find_by(name: smart_proxy)
        raise Exception unless ansible_proxy.has_capability?('Ansible', 'vcs')
```

Reply:
Thank you very much for the detailed explanation!
It does indeed make sense to only have one of the two plugins be dependent on git.
Adding a new capability to denote cloning functionality is also a nice idea.